bin/post_process_sce.R

```diff
@@ -97,33 +97,25 @@ metadata(sce)$min_gene_cutoff <- opt$gene_cutoff
 alt_exp <- opt$adt_name
 if (alt_exp %in% altExpNames(sce)) {
   
-  # check for NAs in filtering columns
-  use_discard <- sum(is.na(altExp(sce, alt_exp)$discard)) == 0
-  use_zero.ambient <- sum(is.na(altExp(sce, alt_exp)$zero.ambient)) == 0
+  # set up filter with all Keep, and then override to Remove where necessary
+  sce$adt_scpca_filter <- "Keep"
+  sce$adt_scpca_filter[which(altExp(sce, alt_exp)$discard)] <- "Remove"
+
+  # Warnings for different types of failure:
+  fail_all_removed <- sum(sce$adt_scpca_filter == "Remove") == length(sce$adt_scpca_filter)
+  fail_all_na <- sum(is.na(altExp(sce, alt_exp)$discard)) == length(sce$adt_scpca_filter)
```

Comment:
Simplify and speed up.
```suggestion
  fail_all_removed <- all(sce$adt_scpca_filter == "Remove")
  fail_all_na <- all(is.na(altExp(sce, alt_exp)$discard)) 
```